Loop/Managers/DeviceDataManager.swift

```diff
@@ -1372,156 +1388,144 @@ extension DeviceDataManager {
             }
         }
         
-        let command: RemoteCommand
+        let action: RemoteAction
         
         do {
-            command = try RemoteCommand.createRemoteCommand(notification: notification, allowedPresets: loopManager.settings.overridePresets, defaultAbsorptionTime: carbStore.defaultAbsorptionTimes.medium).get()
+            action = try RemoteAction.createRemoteAction(notification: notification).get()
         } catch {
             log.error("Remote Notification: Parse Error: %{public}@", String(describing: error))
             return
```

Comment:
There were all sorts of validation logic for specific command types in this method previously. The new paradigm here is to move the validation logic into the RemoteAction. That helps hide this complexity and makes it more easy to unit test.